### PR TITLE
Boyer Moore majority vote Algorithm in c++

### PR DESCRIPTION
**Pull request by rutujadhanawade**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Boyer Moore Algorithm in c++

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1439

### CODE REVIEW

**Comment by rutujadhanawade** [comment]
@iamrajiv Please review this

File: C-Plus-Plus/other/majority_vote_algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// The Boyer–Moore majority vote algorithm in C++
```
Comment (iamrajiv) [inline C-Plus-Plus/other/majority_vote_algorithm.cpp:1]: @rutujadhanawade Add a little bit description of the algorithm.

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/majority_vote_algorithm.cpp (hunk 1 above)
Comment (rutujadhanawade) [reply C-Plus-Plus/other/majority_vote_algorithm.cpp:1]: Done :+1: @iamrajiv

**Review by atarax665 — COMMENTED** [review]
This is not Boyer–Moore majority vote algorithm, this is the general algorithm to find majority element in an array. Please do check your algorithm once.

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@rutujadhanawade In Boyer–Moore majority vote algorithm the time complexity is linear but according to your code it is of quadratic time.

**Comment by rutujadhanawade** [comment]
I took referance from the code present in java folder https://github.com/TesseractCoding/NeoAlgo/blob/master/Java/Algorithms/Moore_Algorithm.java complexity of this algorithm is also not linear.

Now I have made all the changes in my code. Please review.

**Comment by atarax665** [comment]
> I took referance from the code present in java folder https://github.com/TesseractCoding/NeoAlgo/blob/master/Java/Algorithms/Moore_Algorithm.java complexity of this algorithm is also not linear.
> 
> Now I have made all the changes in my code. Please review.

Thank you to let us know about this error.

**Review by atarax665 — APPROVED** [review]

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Comment by rutujadhanawade** [comment]
> Thank you to let us know about this error.

I would like to solve this error , Can I create a new issue for this?

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**